### ISSUE TITLE
Integration isn't accepting the captcha

### ISSUE THREAD

**Issue by drthanwho** [custom-components/alexa_media_player#725]
**Describe the bug**
FIrstly, I have to keep #714 on hold for now since I ended up setting the integration up all over again. Now I have a whole other problem. Hopefully this is easy to fix and then I can swap the python file with your fix to test out. 

So basically after I had that issue I ended up getting it to work by removing integration and re installing it in HACS and setting it up again. Now today I guess it logged me out and it's asking for a captcha. But it's not accepting it.

**To Reproduce**
1. Yes at first picture wouldn't load so I just submitted a made up code to force the new image to show
2. Submitted captcha. Pop up closed but still notification there asking for one 
![image](https://user-images.githubusercontent.com/22405505/82169666-7effeb80-98c2-11ea-8198-5b0bc34dc511.png)
3. Clicking "CONFIGURE" again shows this
![image](https://user-images.githubusercontent.com/22405505/82169629-5d9eff80-98c2-11ea-8122-855ae3ad4719.png)

I've tried 
a. captcha alone
b. Amazon password followed by the captcha
c. Amazon password following by the two factor authentication

I don't know what else I should be trying or if due to the two-factor code it's affecting the login somehow

**System details**
Home-assistant (version): 0.109.2
Hassio (Yes/No): Yes
alexa_media: 2.7.3
alexapy: 1.8.4

**Logs**
```yaml
2020-05-17 22:42:49 ERROR (MainThread) [alexapy.helpers] alexaapi.get_state: A login error occured: An exception of type JSONDecodeError occurred. Arguments:
('Expecting value: line 1 column 1 (char 0)',)
2020-05-17 22:42:50 ERROR (MainThread) [alexapy.helpers] alexaapi.get_state: A login error occured: An exception of type JSONDecodeError occurred. Arguments:
('Expecting value: line 1 column 1 (char 0)',)
2020-05-17 23:21:36 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:22:40 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:23:27 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:23:51 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:25:13 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:27:49 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:27:59 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:28:50 ERROR (MainThread) [alexapy.helpers] alexaapi.get_state: A login error occured: An exception of type JSONDecodeError occurred. Arguments:
('Expecting value: line 1 column 1 (char 0)',)
2020-05-17 23:31:14 ERROR (MainThread) [alexapy.helpers] alexaapi.get_state: A login error occured: An exception of type JSONDecodeError occurred. Arguments:
('Expecting value: line 1 column 1 (char 0)',)
2020-05-17 23:31:15 ERROR (MainThread) [alexapy.helpers] alexaapi.get_state: A login error occured: An exception of type JSONDecodeError occurred. Arguments:
('Expecting value: line 1 column 1 (char 0)',)
2020-05-17 23:31:21 ERROR (MainThread) [alexapy.helpers] alexaapi.get_state: A login error occured: An exception of type JSONDecodeError occurred. Arguments:
('Expecting value: line 1 column 1 (char 0)',)
2020-05-17 23:32:09 ERROR (MainThread) [alexapy.helpers] alexaapi.get_state: A login error occured: An exception of type JSONDecodeError occurred. Arguments:
('Expecting value: line 1 column 1 (char 0)',)
2020-05-17 23:32:13 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:32:14 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:36:15 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:38:21 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:51:09 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-17 23:51:30 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
2020-05-18 03:59:26 ERROR (MainThread) [alexapy.helpers] alexaapi._request:An error occured accessing AlexaAPI: An exception of type AlexapyLoginError occurred. Arguments:
('Unauthorized',)
```

**Additional context**

**Comment by drthanwho** [comment]
After trying a  few more times at first it looked like a second pop up was meant to appear but it was completely empty. I tried again and this time the second pop up gave another captcha and looked like this. 
![image](https://user-images.githubusercontent.com/22405505/82170467-cab39480-98c4-11ea-9f3e-5031fd982f98.png)
This is after I hit confirm and now it comes up but there's no image. Also notice how it now says "Begin"?

Hope this helps. No new logs after doing this by the way

**Comment by alandtse** [comment]
You need to be on the latest version if you want help. There are bug fixes in there.

**Comment by drthanwho** [comment]
Updated restarted and still same issue

**Comment by drthanwho** [comment]
I can't tell if I'm also the issue and not entering the captcha correctly since it's saying it's wrong. After several attempts of some easier captchas I got a different message saying "Incorrect Password" so wondering if it has to do with "2FA" and the integration is providing the password automatically but since it's missing an updated 2FA code it's not signing in?

```yaml
2020-05-18 15:54:21 DEBUG (MainThread) [alexapy.alexalogin] Processing https://www.amazon.co.uk/ap/signin
2020-05-18 15:54:21 DEBUG (MainThread) [alexapy.alexalogin] Error message: There was a problem
            Your password is incorrect
          
2020-05-18 15:54:21 DEBUG (MainThread) [alexapy.alexalogin] Captcha requested
2020-05-18 15:54:21 DEBUG (MainThread) [alexapy.alexalogin] Found post url to https://www.amazon.co.uk/ap/signin
2020-05-18 15:54:21 DEBUG (MainThread) [custom_components.alexa_media.configurator] Testing login status: {'error_message': 'There was a problem\n            Your password is incorrect\n          ', 'captcha_required': True, 'captcha_image_url': '****'}
2020-05-18 15:54:21 DEBUG (MainThread) [custom_components.alexa_media.configurator] Creating configurator to request captcha
2020-05-18 15:54:41 DEBUG (MainThread) [custom_components.alexa_media.configurator] Configurator closed for Status: {'error_message': 'There was a problem\n            Your password is incorrect\n          ', 'captcha_required': True, 'captcha_image_url': *****'}
 got captcha: nnmywd903560 securitycode: None Claimsoption: None AuthSelectOption: None  VerificationCode: None
2020-05-18 15:54:41 DEBUG (MainThread) [alexapy.alexalogin] No valid cookies for log in; using credentials
2020-05-18 15:54:41 DEBUG (MainThread) [alexapy.alexalogin] Loaded last request to https://www.amazon.co.uk/ap/signin 
2020-05-18 15:54:41 DEBUG (MainThread) [alexapy.alexalogin] Processing https://www.amazon.co.uk/ap/signin
2020-05-18 15:54:41 DEBUG (MainThread) [alexapy.alexalogin] Error message: There was a problem
            Your password is incorrect
```

**Comment by alandtse** [comment]
Based on the logs, it's a user input issue.

> Enter the characters as they are shown in the image.

This means your captcha is wrong. I can't help you with this. Please note it is always six digits and does not include you OTP or 2FA. For example `nnmywd903560` is an invalid captcha response always.

2FA is a separate form that occurs after you pass the captcha step.

> Your password is incorrect

Fix your password. You may even want to verify your email to make sure it's right.

**Comment by alandtse** [comment]
You can also your the add integration button and it will guide you through the login process.

**Comment by drthanwho** [comment]
So I get the first ones is me filling in the wrong captcha but there's nothing I can but about the second issue apart from removing the integration and adding it all over again then. This pop up only asks for a captcha and nothing else. 

How can it be my email or password that are wrong when I've already set the integration before this happened

**Comment by alandtse** [comment]
Check your configuration.yaml to make sure it's still accurate. It may make sense to remove it and then install using the Add integration page. You should also try renaming your .pickle file in your `/config/.storage` direcotry.  Also check your email to make sure Amazon isn't forcing you to reset your password.

If you provide the full logs (scrubbed) it may show information I can debug but the excerpts you're showing me show only input errors.

**Comment by drthanwho** [comment]
I don't have an entry in my config yaml. It's all set up from the integrations page.

Those logs were the full logs (with debug on) that I was getting while trying to enter the captcha. 
I don't get what you mean by renaming the pickle file though. Renaming it to what ?

Since it always stops working for me for one reason or another, the only working solution I have found is deleting the integration from the integrations page and adding it again. Doing that I wrote the same email and same password as always and now it's working again (this time it didn't even need a 2FA password or asked for captcha). 

Not sure what this means now
